modules/Bio/EnsEMBL/IO/Parser/BaseVCF4.pm

```diff
@@ -310,17 +310,20 @@ sub get_end {
     my $self = shift;
     my $info = $self->get_info;
     my $end;
+    my $alternatives = join(",", @{$self->get_alternatives});
     if (defined($info->{END})) {
       $end = $info->{END};
     }
     elsif(defined($info->{SVLEN})) {
```

Comment:
@sgiorgetti,

1. The expected change would be to change the order in the if-elsif like so,

currently we have -
```
    if (defined($info->{END})) {
      $end = $info->{END};
    }
    elsif(defined($info->{SVLEN})) {
      return unless $self->get_start;
      my $svlen = (split(',',$info->{SVLEN}))[0];
      $end = $self->get_start + abs($svlen) - 1;
    }
```

change it to -
```
    if (defined($info->{SVLEN})) {
      return unless $self->get_start;
      my $svlen = (split(',',$info->{SVLEN}))[0];
      $end = $self->get_start + abs($svlen) - 1;
    }
    elsif (defined($info->{END})) {
      $end = $info->{END};
    }
```

So we calculate end using SVLEN first if available.

2. Also, for `sub get_raw_end`, it is better to change it to call `get_end` as all these changes should be added in get_raw_end too (but we do not want to duplicate effort) -
```
sub get_raw_end {
    my $self = shift;
    return $self->get_end;
}
```

Reply:
Yes ... and you mentioned that before ... bear with me

Reply:
thanks @sgiorgetti !